### PR TITLE
Add DISABLE_CORS_HEADERS option to disable default CORS headers

### PR DESCRIPTION
**Pull request by whummer**
Add `DISABLE_CORS_HEADERS` option to disable the CORS headers returned from API calls by default. Currently, we're offering several `*_CORS_*` [settings](https://docs.localstack.cloud/localstack/configuration/#security) which are all useful to expand the CORS rules, i.e., make the security _less_ restrictive. However, there have been some requests from users who would like to tighten up the security, i.e., make the CORS settings _more_ restrictive (i.e., disable access from browsers to the local APIs).

Currently, this only affects the default CORS headers we're returning, but **not** the service-specific CORS headers that can be defined on a resource level (e.g., for S3 buckets, or API GW resources). This should be a reasonable assumption for now, but needs to be highlighted in the docs.

TODOs:
- [ ] Add corresponding PR in the docs repo to describe this config value

### CODE REVIEW

File: tests/integration/test_security.py [hunk 1]
```diff
@@ -51,14 +45,14 @@ def test_cors_s3_override(self):
...
             ]
         }
-        bucket_name = "my-s3-bucket"
+        bucket_name = f"bucket-{short_uid()}"
 
         try:
-            client.create_bucket(Bucket=bucket_name)
-            client.put_bucket_cors(Bucket=bucket_name, CORSConfiguration=BUCKET_CORS_CONFIG)
+            aws_stack.get_or_create_bucket(bucket_name)
+            s3_client.put_bucket_cors(Bucket=bucket_name, CORSConfiguration=BUCKET_CORS_CONFIG)
```
Comment (thrau) [inline tests/integration/test_security.py:52]: while we're refactoring this test to pytest, we could maybe also simplify the test by injecting the `s3_bucket` fixture, which gives you the name of a newly created s3 bucket, that is also recursively cleared and deleted after the test run (so we can remove the try/finally block)

**Review by thrau — APPROVED** [review]
LGTM! just a minor nit about the test

**Review by dfangl — APPROVED** [review]
LGTM! :tada:

File: tests/integration/test_security.py (hunk 1 above)
Comment (whummer) [reply tests/integration/test_security.py]: Good catch, thanks! 👍 Updated..

**Pull request merged**